cfdm/read_write/netcdf/netcdfwrite.py

```diff
@@ -4288,8 +4288,8 @@ def file_open(self, filename, mode, fmt, fields):
                 As for the *format* parameter for initialising a
                 `netCDF.Dataset` instance.
 
-            fields: sequence of `Field`
-                The field constructs to be written.
+            fields: sequence of `Field` or `Domain`
+                The constructs to be written.
```

Comment:
I realise you are just extending the description to include `Domain` here, but (in the spirit of raising issues as we see them) actually this parameter description overall seems pretty obscure to me, such that I had to check the code and try to deduce the point here (hoping it's not just a 'Monday Morning' type thing and it is in fact obvious!).

Mainly I was confused why a `file_open` method would be writing anything, as this might imply. So as I understand it from checking the code itself and usage of this, the constructs given in `field` are the ones the user *intends to* (eventually) write out to the file to be opened, though the method doesn't do any writing. I remember we planned to give this code a good refactor / rewrite when we had time, when we could change this to self-contained methods, but for now it would be good to amend the description to imply that sort of thing! Something like:

```suggestion
                The constructs that will at some point be
                written to the file once opened.
```

(Realising the above situation not ideal really as how would someone always know that in advance?)

Assuming I'm not completely mis-understanding this!

Reply:
You are spot on. See https://github.com/NCAS-CMS/cfdm/pull/218/commits/1aed9f4c46b63ceb88cf953c74be94ccc51e83fb